acceptance/suites/tests/030-tk-auth/050-tk-auth.rb

```diff
@@ -0,0 +1,216 @@
+    #TODO: Improve this.
+    if (@rule['type'] == 'regex') &&
+       (@rule['allow'] == '$1') &&
+       (@path.include?('([^/]+)$') ) 
+    then
+      # it is a valid regex rule and we just passed in a raw unprocessed path
+      # so we overwrite path, stripping out the regex
+      # and set url_cn
+      @path = @path.match.('[a-zA-Z0-9\/]+')[0]
+      @url_cn ||= @cn
+    end
+    #TODO: elseif path and url_cn are set then continue as normal.
+  end
+
+
+  def set_defaults
+      @port           ||= '8140'      
+      @method         ||= 'GET'
+      @cacert_file    ||= "/etc/puppetlabs/puppet/ssl/ca/ca_crt.pem"
+      @key_file       ||= "/etc/puppetlabs/puppet/ssl/private_keys/#{@cn}.pem"
+      @hostcert_file  ||= "/etc/puppetlabs/puppet/ssl/certs/#{@cn}.pem"
+      @auth_string    ||= "--cacert #{@cacert_file} --key #{@key_file} --cert #{@hostcert_file}" 
+      @url            ||= "https://" + @host + ':'+ @port + @path
+      @url            += @url_cn unless @url_cn.nil?
+  end
+
+
+  def set_porf
+    @actual_result ||= @actual_result_detail.stdout.match(/httpresponse:.*/)[0].match(/[0-9]../)[0]
+    case @expected_result.class.to_s
+      when 'Array'
+        if @expected_result.include?(@actual_result) then
+          @porf = 'PASS'
+          @@total_passes += 1
+        else
+          @porf = 'FAIL'
+          @@total_fails += 1
+        end
+      when 'String'
+          if @expected_result == @actual_result then 
+            @porf = 'PASS' 
+          else @porf = 'FAIL'
+          @@total_fails += 1
+          end
+      else 
+        puts 'How did we get here?'
+    end
+
+    puts @porf
+  end
+
+
+  def method_match (method_to_test, ruleset_method)
+    # return true if the method in this test matches the method in the selected ruleset
+    ruleset_method = 'GET' if ruleset_method == nil 
+    ruleset_method.include?(method_to_test.downcase)
+  end
+
+
+  def authorized? 
+    # returns true if the rule authorizes the request
+    #TODO: accept a host name and calculate if the rule authorizes the host
+    #we have to handle the case where path contains ([^/]+ and allow => $1 in here.
+    #in this case, we need to seperate 
+    # 1.  the path we are going to test 
+    # 2.  the path in the rule under test
+    # So,
+    # If cn=hostBob is looking for /puppet/v3/report/hostBob Than we AUTH = True
+    # If cn=SomeOtherHost is looking for /puppet/v3/report/hostBob Than AUTH = FALSE.
+    return true if @rule.keys.include?('allow-unauthenticated') && 
+      @rule['allow-unauthenticated'] # (-k and everything else will work)
+
+    if @rule['allow'] != nil then
+      return true if @rule['allow'].include?('*') #TODO: express that -k won't work
+      return true if @rule['allow'].include?('$1') #TODO and its a regex and the cn is signed and the cn is in the url regex thing
+    end
+    return false
+  end
+
+
+  def calculate_expected_result
+    if method_match(@method, @rule['match-request']['method']) &&
```

Comment:
Just an fyi - `method` is not a required field in a rule `match-request`.  We do happen to include it for each rule that we put in our default `auth.conf` file for Puppet Server, though, as you've seen.